--- amissl_glue.c	2014-07-10 11:15:22.691865718 +0200
+++ amissl_glue.c_new	2014-07-10 11:15:12.756099791 +0200
@@ -46,10 +46,12 @@
 
 // ---
 
+#if 0
 long AMISSL_LIB_ENTRY _AmiSSL_InitAmiSSLA(REG(a6, __IFACE_OR_BASE), REG(a0, struct TagItem * tagList))
 {
 	return InitAmiSSLA(tagList);
 }
+#endif
 
 // ---
 
@@ -2328,17 +2330,21 @@
 
 // ---
 
+#if 0
 int AMISSL_LIB_ENTRY _AmiSSL_BIO_vprintf(REG(a6, __IFACE_OR_BASE), REG(a0, BIO * bio), REG(a1, const char * format), REG(a2, VA_LIST args))
 {
 	return BIO_vprintf(bio, format, args);
 }
+#endif
 
 // ---
 
+#if 0
 int AMISSL_LIB_ENTRY _AmiSSL_BIO_vsnprintf(REG(a6, __IFACE_OR_BASE), REG(a0, char * buf), REG(d0, size_t n), REG(a1, const char * format), REG(a2, VA_LIST args))
 {
 	return BIO_vsnprintf(buf, n, format, args);
 }
+#endif
 
 // ---
 
