--- amissl_glue.c_org	2016-03-29 13:27:32.249592414 +0200
+++ amissl_glue.c	2016-03-29 13:28:10.320683798 +0200
@@ -6,6 +6,10 @@
 
 #include "amissl_glue.h"
 
+#include <internal/cryptlib.h>
+#include <internal/o_str.h>
+#include <internal/threads.h>
+
 /***************************************************************************/
 
 #if defined(BASEREL)
@@ -14,9 +17,10 @@
 #undef SAVEDS
 #define SAVEDS __attribute__((baserel_restore)) __attribute__ ((noinline))
 #elif defined(__amigaos3__)
-#error insert the correct offset to let the __restore_a4 function restore the correct base relative data segment below and remove this warning
+//#error insert the correct offset to let the __restore_a4 function restore the correct base relative data segment below and remove this warning
 // restore the base relative data segment from the library base address in register a6
 // this function will be called from all non-static functions in this module if it is built with -mrestore-a4
+#include "amissl_base.h"
 static const USED_VAR unsigned short __restore_a4[] = { 0x286e, OFFSET(LibraryHeader, dataSeg), 0x4e75 }; // "move.l a6@(dataSeg:w),a4;rts"
 #endif // __amigaos3__
 #endif // BASEREL
@@ -30,24 +34,30 @@
 
 // ---
 
+#if 0
 void SAVEDS ASM LIB_InternalInitAmiSSL(REG(a6, UNUSED __IFACE_OR_BASE), REG(a0, void * amisslinit))
 {
 	InternalInitAmiSSL(amisslinit);
 }
+#endif
 
 // ---
 
+#if 0
 long SAVEDS ASM LIB_InitAmiSSLA(REG(a6, UNUSED __IFACE_OR_BASE), REG(a0, struct TagItem * tagList))
 {
 	return InitAmiSSLA(tagList);
 }
+#endif
 
 // ---
 
+#if 0
 long SAVEDS ASM LIB_CleanupAmiSSLA(REG(a6, UNUSED __IFACE_OR_BASE), REG(a0, struct TagItem * tagList))
 {
 	return CleanupAmiSSLA(tagList);
 }
+#endif
 
 // ---
 
