--- amisslext_m68k.c	2022-03-21 20:31:25.159124470 +0000
+++ amisslext_m68k.c_patched	2022-03-21 20:31:02.203923837 +0000
@@ -38,6 +38,7 @@
 
 /***************************************************************************/
 
+#if 0
 STATIC struct Library * stub_ext_Open_PPC(uint32 *regarray)
 {
 	struct Library *Base = (struct Library *)regarray[REG68K_A6/4];
@@ -75,6 +76,7 @@
 	return 0;
 }
 STATIC CONST struct EmuTrap stub_ext_Reserved = { TRAPINST, TRAPTYPE, (uint32 (*)(uint32 *))stub_ext_Reserved_PPC };
+#endif
 
 // ---
 
@@ -14752,10 +14754,10 @@
 
 CONST CONST_APTR ext_VecTable68K[] =
 {
-	&stub_ext_Open,
-	&stub_ext_Close,
-	&stub_ext_Expunge,
-	&stub_ext_Reserved,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
 	&stub_ext_OSSL_trace_enabled,
 	&stub_ext_OSSL_trace_begin,
 	&stub_ext_OSSL_trace_end,
