--- amissl_glue.c_old	2014-06-22 18:18:58.427031368 +0200
+++ amissl_glue.c	2014-06-22 21:25:51.335617507 +0200
@@ -25,45 +25,57 @@
 
 /***************************************************************************/
 
+#if 0
 ULONG AMISSL_LIB_ENTRY _AmiSSL_Obtain(REG(a6, __IFACE_OR_BASE))
 {
 	return Obtain();
 }
+#endif
 
 // ---
 
+#if 0
 ULONG AMISSL_LIB_ENTRY _AmiSSL_Release(REG(a6, __IFACE_OR_BASE))
 {
 	return Release();
 }
+#endif
 
 // ---
 
+#if 0
 void AMISSL_LIB_ENTRY _AmiSSL_InternalInitAmiSSL(REG(a6, __IFACE_OR_BASE), REG(a0, void * amisslinit))
 {
 	InternalInitAmiSSL(amisslinit);
 }
+#endif
 
 // ---
 
+#if 0
 long AMISSL_LIB_ENTRY _AmiSSL_InitAmiSSLA(REG(a6, __IFACE_OR_BASE), REG(a0, struct TagItem * tagList))
 {
 	return InitAmiSSLA(tagList);
 }
+#endif
 
 // ---
 
+#if 0
 long AMISSL_LIB_ENTRY _AmiSSL_CleanupAmiSSLA(REG(a6, __IFACE_OR_BASE), REG(a0, struct TagItem * tagList))
 {
 	return CleanupAmiSSLA(tagList);
 }
+#endif
 
 // ---
 
+#if 0
 long AMISSL_LIB_ENTRY _AmiSSL_IsCipherAvailable(REG(a6, __IFACE_OR_BASE), REG(d0, long cipher))
 {
 	return IsCipherAvailable(cipher);
 }
+#endif
 
 // ---
 
@@ -2391,17 +2403,21 @@
 
 // ---
 
+#ifndef __amigaos4__
 int AMISSL_LIB_ENTRY _AmiSSL_BIO_vprintf(REG(a6, __IFACE_OR_BASE), REG(a0, BIO * bio), REG(a1, const char * format), REG(a2, long * args))
 {
 	return BIO_vprintf(bio, format, args);
 }
+#endif
 
 // ---
 
+#ifndef __amigaos4__
 int AMISSL_LIB_ENTRY _AmiSSL_BIO_vsnprintf(REG(a6, __IFACE_OR_BASE), REG(a0, char * buf), REG(d0, size_t n), REG(a1, const char * format), REG(a2, long * args))
 {
 	return BIO_vsnprintf(buf, n, format, args);
 }
+#endif
 
 // ---
 
@@ -10357,24 +10373,30 @@
 
 // ---
 
+#if 0
 const SSL_METHOD * AMISSL_LIB_ENTRY _AmiSSL_SSLv23_method(REG(a6, __IFACE_OR_BASE))
 {
 	return SSLv23_method();
 }
+#endif
 
 // ---
 
+#if 0
 const SSL_METHOD * AMISSL_LIB_ENTRY _AmiSSL_SSLv23_server_method(REG(a6, __IFACE_OR_BASE))
 {
 	return SSLv23_server_method();
 }
+#endif
 
 // ---
 
+#if 0
 const SSL_METHOD * AMISSL_LIB_ENTRY _AmiSSL_SSLv23_client_method(REG(a6, __IFACE_OR_BASE))
 {
 	return SSLv23_client_method();
 }
+#endif
 
 // ---
 
