--- amissl_glue.c	2014-06-10 14:57:34.595406066 +0200
+++ amissl_glue.c	2014-06-10 14:50:11.413688631 +0200
@@ -25,6 +25,7 @@
 
 /***************************************************************************/
 
+#if 0
 ULONG AMISSL_LIB_ENTRY _AmiSSL_Obtain(REG(a6, __IFACE_OR_BASE))
 {
 	return Obtain();
@@ -64,6 +65,7 @@
 {
 	return IsCipherAvailable(cipher);
 }
+#endif
 
 // ---
 
@@ -2391,17 +2393,21 @@
 
 // ---
 
+#ifndef __amigaos4__
 int AMISSL_LIB_ENTRY _AmiSSL_BIO_vprintf(REG(a6, __IFACE_OR_BASE), REG(a0, BIO * bio), REG(a1, const char * format), REG(a2, long * args))
 {
 	return BIO_vprintf(bio, format, args);
 }
+#endif
 
 // ---
 
+#ifndef __amigaos4__
 int AMISSL_LIB_ENTRY _AmiSSL_BIO_vsnprintf(REG(a6, __IFACE_OR_BASE), REG(a0, char * buf), REG(d0, size_t n), REG(a1, const char * format), REG(a2, long * args))
 {
 	return BIO_vsnprintf(buf, n, format, args);
 }
+#endif
 
 // ---
 
@@ -4687,10 +4693,12 @@
 
 // ---
 
+#ifndef __amigaos4__
 void AMISSL_LIB_ENTRY _AmiSSL_ERR_add_error_dataA(REG(a6, __IFACE_OR_BASE), REG(d0, int num), REG(a0, long * args))
 {
 	ERR_add_error_dataA(num, args);
 }
+#endif
 
 // ---
 
@@ -10371,24 +10379,30 @@
 
 // ---
 
+#if 0
 SSL_METHOD * AMISSL_LIB_ENTRY _AmiSSL_SSLv23_method(REG(a6, __IFACE_OR_BASE))
 {
 	return SSLv23_method();
 }
+#endif
 
 // ---
 
+#if 0
 SSL_METHOD * AMISSL_LIB_ENTRY _AmiSSL_SSLv23_server_method(REG(a6, __IFACE_OR_BASE))
 {
 	return SSLv23_server_method();
 }
+#endif
 
 // ---
 
+#if 0
 SSL_METHOD * AMISSL_LIB_ENTRY _AmiSSL_SSLv23_client_method(REG(a6, __IFACE_OR_BASE))
 {
 	return SSLv23_client_method();
 }
+#endif
 
 // ---
 
